spark/src/main/java/org/apache/iceberg/spark/SparkSchemaUtil.java

```diff
@@ -63,6 +65,21 @@ public static Schema schemaForTable(SparkSession spark, String name) {
     return new Schema(converted.asNestedType().asStructType().fields());
   }
 
+  /**
+   * Given a Spark table identifier, determine the PartitionSpec.
+   * @param spark the SparkSession which contains the identifier
+   * @param table a TableIdentifier, if the namespace is left blank the catalog().currentDatabase() will be used
+   * @return a IcebergPartitionSpec representing the partitioning of the Spark table
+   * @throws AnalysisException if thrown by the Spark catalog
+   */
+  public static PartitionSpec specForTable(SparkSession spark, TableIdentifier table) throws AnalysisException {
```

Comment:
I don't think that we should leak the v1 Spark API (`TableIdentifier`) in a util class like this. What about passing database and name separately? Then the caller is responsible for adding database, which avoids the need to use v1 `spark.catalog()` to default it.

Reply:
Sure, the main issue is that the old version just always used the string literal "default" no matter what. Which I don't think is the right thing to do.

Reply:
I actually am not using this method any more so I'll drop it and revert back to "default in the base specForTable